samples/webserver/index.js

```diff
@@ -29,6 +31,42 @@ onload = function() {
 
   var logToScreen = function(log) {
     logger.textContent += log + "\n";
+    logger.scrollTop = logger.scrollHeight;
+  }
+
+  var destroySocketById = function(socketId) {
+    tcpSocket.disconnect(socketId,function(){
+      tcpSocket.close(socketId);
+    });
+  }
+
+  var closeServerSocket = function() {
+    if(serverSocketId) {
+      tcpServer.close(serverSocketId);
+    }
+
+    tcpServer.onAccept.removeListener(onAccept);
+    tcpSocket.onReceive.removeListener(onReceive);
+  }
+
+  var sendReplyToSocket = function(socketId,buffer,keepAlive) {
+    // verify that socket is still connected before trying to send data
+    tcpSocket.getInfo(socketId,function(socketInfo) {
+      if(socketInfo.connected !== true) {
+        destroySocketById(socketId);
+        return;
+      }
+
+      tcpSocket.setKeepAlive(socketId, keepAlive, function() {
```

Comment:
I set the delay to 1 and I'm checking for `chrome.runtime.lastError`, if an error occurs I'm aborting the `send` and closing the socket. Not sure if this is the right way to do it, or we could still call `send` after `setKeepAlive` fails. (couldn't reproduce on Canary to properly test this)